lib/DiscoveryStreamFeed.jsm

```diff
@@ -1182,6 +1182,21 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
           const filtered = spocsList.filter(s => s.url === action.data.url);
           if (filtered.length) {
             this._sendSpocsFill({ blocked_by_user: filtered }, false);
+
+            // If we're blocking a spoc, we want a slightly different treatment for open tabs.
+            this.store.dispatch(
+              ac.AlsoToPreloaded({
```

Comment:
why we are not using BroadcastToContent for spoc dismissal? How AlsoToPreloaded differs from BroadcastToContent

Reply:
AlsoToPreloaded got to just main and preloaded, BroadcastToContent go to everything.

So in this case, everything gets the `DISCOVERY_STREAM_SPOC_BLOCKED` event, but existing tabs do not get the `DISCOVERY_STREAM_LINK_BLOCKED` event.

Reply:
To be more clear, if action.data is undefined, both events fail in the same way, when they try to access `action.data.url`, and hit a js error. If action.data is there, but no url, they both fail at doing their job but shouldn't throw a js error.

Both cases are currently unexpected to happen.

Reply:
Actually, it would fail even sooner on `const filtered = spocsList.filter(s => s.url === action.data.url);` above in this function.

Also just noticed I commented to the wrong comment.

Reply:
cool, and thats because we don't want blocked spoc to be filtered out from prevState.spocs.data.spocs , so that we can use spocs state to find blocked spoc position in selectlayoutrender.

Reply:
Correct.